tests/Integration/Momento.Sdk.Tests/TopicTest.cs

```diff
@@ -0,0 +1,236 @@
+    }
+
+    [Fact]
+    public async Task PublishAsync_BadCacheNameByteArray_IsError()
+    {
+        var response = await topicClient.PublishAsync("fake-" + cacheName, "topic", Array.Empty<byte>());
+        Assert.True(response is TopicPublishResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.NOT_FOUND_ERROR, ((TopicPublishResponse.Error)response).ErrorCode);
+    }
+
+    [Theory]
+    [InlineData(null, "topic")]
+    [InlineData("cache", null)]
+    public async Task PublishAsync_NullChecksString_IsError(string badCacheName, string badTopicName)
+    {
+        var response = await topicClient.PublishAsync(badCacheName, badTopicName, "value");
+        Assert.True(response is TopicPublishResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((TopicPublishResponse.Error)response).ErrorCode);
+    }
+
+    [Fact]
+    public async Task PublishAsync_PublishNullString_IsError()
+    {
+        var response = await topicClient.PublishAsync(cacheName, "topic", (string)null!);
+        Assert.True(response is TopicPublishResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((TopicPublishResponse.Error)response).ErrorCode);
+    }
+
+    [Fact]
+    public async Task PublishAsync_BadCacheNameString_IsError()
+    {
+        var response = await topicClient.PublishAsync("fake-" + cacheName, "topic", "value");
+        Assert.True(response is TopicPublishResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.NOT_FOUND_ERROR, ((TopicPublishResponse.Error)response).ErrorCode);
+    }
+
+    [Theory]
+    [InlineData(null, "topic")]
+    [InlineData("cache", null)]
+    public async Task SubscribeAsync_NullChecks_IsError(string badCacheName, string badTopicName)
+    {
+        var response = await topicClient.SubscribeAsync(badCacheName, badTopicName);
+        Assert.True(response is TopicSubscribeResponse.Error, $"Unexpected response: {response}");
+        Assert.Equal(MomentoErrorCode.INVALID_ARGUMENT_ERROR, ((TopicSubscribeResponse.Error)response).ErrorCode);
+    }
+
+    [Fact(Timeout = 5000)]
+    public async Task PublishAndSubscribe_ByteArray_Succeeds()
+    {
+        const string topicName = "topic_bytes";
+        var skipMessage = new byte[] { 0x00 };
+        var valuesToSend = new List<byte[]>
+        {
+            new byte[] { 0x01 },
+            new byte[] { 0x02 },
+            new byte[] { 0x03 },
+            new byte[] { 0x04 },
+            new byte[] { 0x05 }
+        };
+
+        var subscribeResponse = await topicClient.SubscribeAsync(cacheName, topicName);
+        Assert.True(subscribeResponse is TopicSubscribeResponse.Subscription,
+            $"Unexpected response: {subscribeResponse}");
+        var subscription = (TopicSubscribeResponse.Subscription)subscribeResponse;
```

Comment:
Yeah, casting being the pattern is not great. I think you should consider:
* Make TopicSubscribeResponse the return type.
* Make TopicSubscribeResponse a simple class.
* Un-nest response types.
* Compose 2 things in the TopicSubscribeResponse:
  1. An enum `TopicSubscribeResponse.Kind`
  2. A field for each response kind (like protocol buffers, sound familiar?)
* Add a couple properties to TopicSubscribeResponse:
  1. Kind - documented so people can choose to switch on it if they want to avoid the throws below...
  2. Subscription - gives you the Subscription or throws a good error about the error.
  3. Error - gives you the Error or throws a good error about how this was not an error.

Reply:
Casting is not the normal pattern outside of testing. In use it would look more like:
```
if (subscribeResponse is TopicSubscribeResponse.Subscription subscription)
{
      await foreach (var message in subscription)
      {
          switch (message)
          {
              case TopicMessage.Binary binary:
                  break;
              case TopicMessage.Text text:
                  break;
              case TopicMessage.Discontinuity discontinuity:
                  break;
              case TopicMessage.Error:
                  break;
          }
      }
} else if (subscribeResponse is TopicSubscribeResponse.Error error)
{
      // error handling
}
```
That's assuming we split text and binary messages and add discontinuity.

Reply:
@cprice404 Is there a good place to add an example like this without releasing and adding it to the examples directory? I can make something separate if we want realistic looking code before the examples are ready.

Reply:
Oh yeah, that's pretty cool. It looks like a type switch + lexical cast. I couldn't tell that's how it was going :-). Flattening the value cases seems reasonable to me too.

Reply:
re: how to show examples, generally I will just add a throwaway file with the example code, and remove it before the PR is merged. Or add a temporary dependency on `../blah` in the Examples project and include an example there, and stash it before the final PR merge.

or pasting an example into github like you did is fine too.

Reply:
The rewritten tests have a consume function that looks much more like what standard use is: https://github.com/momentohq/client-sdk-dotnet/blob/9ad31e5454674c670913f3feca68d64eddd7cf11/tests/Integration/Momento.Sdk.Tests/TopicTest.cs#L171